messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -146,6 +173,106 @@ public boolean scheduleRetry(CommandMessage commandMessage,
         public B retryExecutor(ScheduledExecutorService retryExecutor) {
             assertNonNull(retryExecutor, "ScheduledExecutorService may not be null");
             this.retryExecutor = retryExecutor;
+
+            //noinspection unchecked
+            return (B) this;
+        }
+
+        /**
+         * Resets the {@link Predicate} chain used for checking non-transiency of a failure. Throws away any previously
+         * existing predicate chain and replaces it with the provided predicate.
+         * <p/>
+         * Provided {@code nonTransientFailurePredicate} checks whether a failure is transient (predicate returns
+         * {@code false}) or not (predicate returns {@code true}). Non-transient failures are never retried.
+         *
+         * @param nonTransientFailurePredicate a {@link Predicate} (accepting a {@link Throwable} parameter) used for
+         *                                     testing transiency of provided failure
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public B nonTransientFailurePredicate(Predicate<Throwable> nonTransientFailurePredicate) {
+            assertNonNull(nonTransientFailurePredicate, "Predicate may not be null");
+            this.nonTransientFailurePredicate = nonTransientFailurePredicate;
+
+            // noinspection unchecked
+            return (B) this;
+        }
+
+        /**
+         * Resets the {@link Predicate} chain used for checking non-transiency of a failure. Throws away any previously
+         * existing predicate chain and replaces it with the provided predicate.
+         * <p/>
+         * Provided {@code nonTransientFailurePredicate} checks whether a failure is transient (predicate returns
+         * {@code false}) or not (predicate returns {@code true}). Non-transient failures are never retried.
+         * <p/>
+         * In essence, this is just a failure typed variant of {@link #nonTransientFailurePredicate(Predicate)} method.
+         *
+         * @param failureType concrete class of the failure to handle
+         * @param nonTransientFailurePredicate a {@link Predicate} (accepting a {@code failureType} parameter) used
+         *                                     for testing transiency of provided failure
+         * @param <E> type of the failure to handle
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public <E extends Throwable> B nonTransientFailurePredicate(Class<E> failureType, Predicate<? super E> nonTransientFailurePredicate) {
+            assertNonNull(failureType, "Class of failure type may not be null");
+            assertNonNull(nonTransientFailurePredicate, "Predicate may not be null");
```

Comment:
I'd be a little more specific in the exception message, stating that the `"Non-transient failure predicate may not be null"`